### ISSUE TITLE
error while training...

### ISSUE THREAD

**Issue by innocent-oss** [fenglinglwb/MAT#60]
![image](https://user-images.githubusercontent.com/71064498/215659837-0ee4463a-ea30-48f9-a69a-78ee45f8fc41.png)

**Comment by newprogrammer2022** [comment]
The same error occured in my traing. I don't unstand why it happened. I tested the loading data filenames and found that when loading the val data , the programe generate raw_idx for val_images , and the first val_images in dir was loaded twice , so the raw_idx is one more than the real val_images amounts.  
I also look foward to the answer by the author.
Thanks very much.

**Comment by fenglinglwb** [comment]
If you assign the right metric to use 2993 images, there shouldn't be an index of 2993 (0-2992). You may define a proper metric by yourself in metric_main.py. Hope this helps.

**Comment by innocent-oss** [comment]
> If you assign the right metric to use 2993 images, there shouldn't be an index of 2993 (0-2992). You may define a proper metric by yourself in metric_main.py. Hope this helps.

I have 1 GPU and using batch 4 it's too slow any solution? I thought batch 32 won't be possible for 1 GPU that's y I changed it to 4

**Comment by newprogrammer2022** [comment]
Did you solve the problem? I tried use different metric like fid50k or fid36k5_full , but  this error always happened ,no matter how many val_images used.  Can you tell me how do you set the parameter if you successed?
Looking forward to you answer/

**Comment by innocent-oss** [comment]
> Did you solve the problem? I tried use different metric like fid50k or fid36k5_full , but this error always happened ,no matter how many val_images used. Can you tell me how do you set the parameter if you successed? Looking forward to you answer/

yeah i used 2993 metric and it works but it's too heavy for me to run

**Comment by newprogrammer2022** [comment]
Fine , may I ask how do you print in command line?  I use : python train.py --outdir=new_model --gpus=1 --batch=2 --data=asian-stars-train/images --data_val=asian-stars-train/val_imgs --mirror=False --metrics=fid2993_full
But still failed in same error?  lt is so werid.

**Comment by innocent-oss** [comment]
> Fine , may I ask how do you print in command line? I use : python train.py --outdir=new_model --gpus=1 --batch=2 --data=asian-stars-train/images --data_val=asian-stars-train/val_imgs --mirror=False --metrics=fid2993_full But still failed in same error? lt is so werid.

I was using colab pro but faced runtime issue after 24hours so still confused what to do

**Comment by yaojingchi** [comment]
I also face the problem too, just keep saying "index N is out of bounds for axis 0 with size N", how to solve this problem

**Comment by nimaone** [comment]
did anyone find solution for this error?

**Comment by dwkim78** [comment]
It seems that one need to create a new function such as "fidXXXX_full(opts)" and register it, where XXXX is the number of samples in the validation dataset. 

You can find lots of "fidXXXX_full(opts)" examples in "metrics/metric_main.py".  You also need to change "max_real" and "num_gen" accordingly. 

Frankly speaking, I do not get it why so many "fidXXXX_full(opts)" functions are necessary in the original code. Anyway I solved the problem in this way.

Hope this helps.

**Comment by Michaelwjh** [comment]
> It seems that one need to create a new function such as "fidXXXX_full(opts)" and register it, where XXXX is the number of samples in the validation dataset.
> 
> You can find lots of "fidXXXX_full(opts)" examples in "metrics/metric_main.py". You also need to change "max_real" and "num_gen" accordingly.
> 
> Frankly speaking, I do not get it why so many "fidXXXX_full(opts)" functions are necessary in the original code. Anyway I solved the problem in this way.
> 
> Hope this helps.

could you share the code? thanks a lot.

**Comment by dwkim78** [comment]
> > It seems that one need to create a new function such as "fidXXXX_full(opts)" and register it, where XXXX is the number of samples in the validation dataset.
> > You can find lots of "fidXXXX_full(opts)" examples in "metrics/metric_main.py". You also need to change "max_real" and "num_gen" accordingly.
> > Frankly speaking, I do not get it why so many "fidXXXX_full(opts)" functions are necessary in the original code. Anyway I solved the problem in this way.
> > Hope this helps.
> 
> could you share the code? thanks a lot.

When there are 500 validation samples, put the following code to "metrics/metric_main.py".

<img width="693" alt="Screenshot 2024-07-04 at 12 40 43 PM" src="https://github.com/fenglinglwb/MAT/assets/7696163/8b6e67e6-e264-4855-8045-56ee932de71f">

Then use 'fid_custom' for the "--metrics" keyword while training using "train.py"

Hope this helps,

**Comment by Michaelwjh** [comment]
> > > It seems that one need to create a new function such as "fidXXXX_full(opts)" and register it, where XXXX is the number of samples in the validation dataset.
> > > You can find lots of "fidXXXX_full(opts)" examples in "metrics/metric_main.py". You also need to change "max_real" and "num_gen" accordingly.
> > > Frankly speaking, I do not get it why so many "fidXXXX_full(opts)" functions are necessary in the original code. Anyway I solved the problem in this way.
> > > Hope this helps.
> > 
> > 
> > could you share the code? thanks a lot.
> 
> When there are 500 validation samples, put the following code to "metrics/metric_main.py".
> 
> <img alt="Screenshot 2024-07-04 at 12 40 43 PM" width="693" src="https://private-user-images.githubusercontent.com/7696163/345688976-8b6e67e6-e264-4855-8045-56ee932de71f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzMTMyMDMsIm5iZiI6MTcyMDMxMjkwMywicGF0aCI6Ii83Njk2MTYzLzM0NTY4ODk3Ni04YjZlNjdlNi1lMjY0LTQ4NTUtODA0NS01NmVlOTMyZGU3MWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDdUMDA0MTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTQwMGQ1NWQ4NmI1ZmM2YTRjY2UzNzgwYTA1Y2NhMGM5MzQxOGEzNmI2MzQzMjVhMjY4NmFmNTg0YTA5ZmU3NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.4d_TJEhezevGwhqopFhGZkXs-1iNYuqB2yvqB4UpBic">
> Then use 'fid_custom' for the "--metrics" keyword while training using "train.py"
> 
> Hope this helps,

Thank you
